### PR TITLE
support brpc baidu-std protocol

### PR DESCRIPTION
**Pull request by smwyzi**

### CODE REVIEW

**Comment by zhaohuabing** [comment]
@smwyzi It would be useful to validate the trpc codec if we can add testing scripts in the test directory.

Are you planning to add the test in another PR?

**Comment by smwyzi** [comment]
> @smwyzi It would be useful to validate the trpc codec if we can add testing scripts in the test directory.
> 
> Are you planning to add the test in another PR?

yes, i will add test case soon

File: src/application_protocols/brpc/protocol.cc [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#include "src/application_protocols/brpc/protocol.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace MetaProtocolProxy {
+namespace Brpc {
+
+const uint32_t BrpcHeader::HEADER_SIZE = 12;
+const uint32_t BrpcHeader::MAGIC = *static_cast<uint32_t*>(static_cast<void*>((const_cast<char*>("PRPC"))));
```
Comment (zhaohuabing) [inline src/application_protocols/brpc/protocol.cc:10]: PRPC -> BRPC ?

**Review by zhaohuabing — APPROVED** [review]
LGTM

**Pull request merged**